--- a/src/maxima.system
+++ b/src/maxima.system
@@ -48,6 +48,12 @@
 (export '*maxima-build-time*)
 
 #+ecl
+(require :cmp)
+
+#+ecl
+(setf c::*compile-in-constants* t)
+
+#+ecl
 (defun do-compile-ecl (file &rest args)
   ;; Not sure about this.  Compile twice.  The first produces a .o
   ;; file.  The second produces the .fas file to keep defsystem happy
@@ -75,6 +81,11 @@
 			     ;; Convert dir/foo.fas to dir/foo.o
 			     (make-pathname :type "o" :defaults p))
 			 files)))
+	(c::build-fasl "binary-ecl/maxima" :lisp-files obj
+			  :ld-flags
+			  (let ((x (symbol-value (find-symbol "*AUTOCONF-LD-FLAGS*"
+							      (find-package "MAXIMA")))))
+			    (if (and x (not (string= x ""))) (list x))))
 	(c::build-fasl "binary-ecl/maxima" :lisp-files obj
 			  :ld-flags
 			  (let ((x (symbol-value (find-symbol "*AUTOCONF-LD-FLAGS*"

